atomic_reactor/utils/operator.py

```diff
@@ -734,6 +734,47 @@ def _replace_pullspecs_not_in_annotations(self, obj, k_or_i, replacements):
             # in replacement_pullspecs and no replacement will be done
             self._replace_unnamed_pullspec(obj, k_or_i, replacements)
 
+    def modifications_append(self, append_mods):
+        """Append to a list entry in `self.data` (or add a list, if one does not exist)
+
+        Also checks the result against schema.
+
+        :param append_mods: a dict
+        :return: None; this modifies 'self.data' in-place
+        """
+        if not append_mods:
+            return
+        for mod in append_mods:
+            for key in append_mods[mod]:
+                item = append_mods[mod][key]
+                if isinstance(item, list):
+                    self.data[mod].setdefault(key, []).extend(item)
+                else:
+                    raise RuntimeError(f'Expected a list, got "{type(item)}"')
+        check_csv(self.data, _mini_csv_schema)
+
+    def modifications_update(self, update_mods):
+        """Update a dict entry in `self.data` (or add a dict entry, if one does not exist)
+
+        Also checks the result against schema.
+
+        :param update_mods: a dict
+        :return: None; this modifies 'self.data' in-place
+        """
+        if not update_mods:
+            return
+
+        def deep_update(target, updates):
+            for key, value in updates.items():
+                if isinstance(value, dict):
+                    # tldr; recursive
+                    target[key] = deep_update(target.get(key, {}), value)
+                else:
+                    target[key] = value
+            return target
+        deep_update(self.data, update_mods)
```

Comment:
can we have a newline here?

Reply:
@MartinBasti; Before the call to `deep_update()`, or after...?

Reply:
before